### PR TITLE
added quaternion methods for inverse and vector rotation

### PR DESCRIPTION
**Pull request by tumbili**
Still have to write unit test for this.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
The unit test doesn't pass any more - odd.

**Comment by tumbili** [comment]
Are you sure? I got this mail that all tests have passed.

File: src/lib/mathlib/math/Quaternion.hpp [hunk 1]
```diff
@@ -116,6 +116,35 @@ class __EXPORT Quaternion : public Vector<4>
...
+	*/
+	Vector<3> rotate(const Vector<3> &w) {
+		Quaternion q_w;	// extend vector to quaternion
+		Quaternion q = {data[0],data[1],data[2],data[3]};
+		Quaternion q_rotated;	// quaternion representation of rotated vector
+		q_w(0) = 0;
+		q_w(1) = w.data[0];
+		q_w(2) = w.data[1];
+		q_w(3) = w.data[2];
+		q_rotated = q*q_w*q.inverse();
```
Comment (thomasgubler) [inline src/lib/mathlib/math/Quaternion.hpp]: You can save some operations if you multiply this out. q_rotated[0] will always be 0. https://github.com/mavlink/mavlink/blob/master/pymavlink/quaternion.py#L169

Comment (thomasgubler) [inline src/lib/mathlib/math/Quaternion.hpp]: Also other terms cancel if I remember correctly

Comment (tumbili) [inline src/lib/mathlib/math/Quaternion.hpp]: Thanks, will have a look at it.

**Pull request merged**